frontend/src/components/Map.vue

```diff
@@ -246,6 +250,11 @@ export default {
 
         //Submit locations to background for optimization
         submitLocations() {
+            //This removes all the elements in the map that were deleted (became null)
```

Comment:
Because the backend code relies on the position of elements added to the coordinate arrays, we cannot simply delete entries because this would mess up the order. Therefore, we instead null the item that was deleted before submission and then use this function to remove the null at the end to maintain proper ordering.